tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -419,17 +427,27 @@ Status GetTrtBroadcastShape(const TRT_TensorOrWeights& operand_l,
       // be broadcasted.
       output_dims_array[0] = -1;
     }
-    // Copy to output dimensions (stripping the batch dimension).
-    output_dims->nbDims = broadcast_num_dims - 1;
-    std::copy(output_dims_array + 1, output_dims_array + broadcast_num_dims,
-              output_dims->d);
+    // Copy to output dimensions
+    if (use_implicit_batch) {
+      // Strip batch dimension while copying
+      output_dims->nbDims = broadcast_num_dims - 1;
+      std::copy(output_dims_array + 1, output_dims_array + broadcast_num_dims,
+                output_dims->d);
+    } else {
+      output_dims->nbDims = broadcast_num_dims;
+      std::copy(output_dims_array, output_dims_array + broadcast_num_dims,
+                output_dims->d);
+    }
+
     return Status::OK();
   };
 
   // Compute the output dimensions.
   const int broadcast_num_dims =
-      std::max(operand_l.GetTrtDims().nbDims + (operand_l.is_tensor() ? 1 : 0),
-               operand_r.GetTrtDims().nbDims + (operand_r.is_tensor() ? 1 : 0));
+      std::max(operand_l.GetTrtDims().nbDims +
+                   (use_implicit_batch && (operand_l.is_tensor() ? 1 : 0)),
```

Comment:
Should the condition be `(use_implicit_batch && operand_l.is_tensor())`?

Reply:
I don't think there is a difference in the final result?

I just simplified it to the following.
Let me know if you think this is better:

```
      std::max(operand_l.GetTrtDims().nbDims + 
                   use_implicit_batch && operand_l.is_tensor(),
               operand_r.GetTrtDims().nbDims +
                   use_implicit_batch && operand_r.is_tensor());
```